course-mw/enrolment-actor/src/main/scala/org/sunbird/group/GroupAggregatesActor.scala

```diff
@@ -0,0 +1,144 @@
+package org.sunbird.group
+
+import org.apache.commons.collections.CollectionUtils
+import org.apache.commons.collections.MapUtils
+import org.apache.commons.lang3.StringUtils
+import org.sunbird.common.exception.ProjectCommonException
+import org.sunbird.common.models.response.Response
+import org.sunbird.common.models.util.{ProjectLogger}
+import org.sunbird.common.request.Request
+import org.sunbird.common.responsecode.ResponseCode
+import org.sunbird.keys.SunbirdKey
+import org.sunbird.learner.actors.group.dao.impl.GroupDaoImpl
+import org.sunbird.actor.base.BaseActor
+import org.sunbird.cache.CacheFactory
+import org.sunbird.cache.interfaces.Cache
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+
+class GroupAggregatesActor extends BaseActor {
+
+
+  private val GROUP_MEMBERS_METADATA: java.util.List[String] = java.util.Arrays.asList("name", "userId", "role", "status", "createdBy")
+
+
+  var groupDao: GroupDaoImpl = new GroupDaoImpl()
+  var groupAggregatesUtil: GroupAggregatesUtil = new GroupAggregatesUtil()
+  var redisCache: Cache = CacheFactory.getInstance()
+  val ttl: Long = 3600
+
+  @throws[Throwable]
+  override def onReceive(request: Request): Unit = {
+    request.getOperation match {
+      case "groupActivityAggregates" => getGroupActivityAggregates(request)
+      case _ => ProjectCommonException.throwClientErrorException(ResponseCode.invalidRequestData,
+        ResponseCode.invalidRequestData.getErrorMessage)
+    }
+  }
+
+  def getCacheKey(groupId: String, activityId: String, activityType: String) = {
+    groupId + ":" + activityId + ":" + activityType + ":activity-agg"
+  }
+
+  def getGroupActivityAggregates(request: Request): Unit = {
+    val groupId: String = request.get(SunbirdKey.GROUPID).asInstanceOf[String]
+    val activityId: String = request.get(SunbirdKey.ACTIVITYID).asInstanceOf[String]
+    val activityType: String = request.get(SunbirdKey.ACTIVITYTYPE).asInstanceOf[String]
+
+    try {
+      val key = getCacheKey(groupId, activityId, activityType)
+      val cachedResponse = redisCache.get("activity-agg", key, classOf[Response])
+      val response = {
+        if(null != cachedResponse) {
+          cachedResponse
+        } else {
+          val memberList: java.util.List[java.util.Map[String, AnyRef]] = getGroupMember(groupId, request)
+          val enrolledGroupMember: java.util.List[java.util.Map[String, AnyRef]] = getEnrolledGroupMembers(activityId, activityType, memberList)
+          populateResponse(groupId, activityId, activityType, enrolledGroupMember, memberList)
+        }
+      }
+      sender().tell(response, self)
```

Comment:
The code can be lot cleaner. We can do like this:
Option(response) match {
    case None => fetchFromDB()
    case Some => response
}